pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1643,6 +1643,11 @@ func NewVolumeBindingPredicate(binder *volumebinder.VolumeBinder) FitPredicate {
 }
 
 func (c *VolumeBindingChecker) predicate(pod *v1.Pod, meta PredicateMetadata, nodeInfo *schedulernodeinfo.NodeInfo) (bool, []PredicateFailureReason, error) {
+	// If pod does not request any volumes, we don't need to do anything.
+	if len(pod.Spec.Volumes) == 0 {
```

Comment:
This might help the benchmark but I'm not sure if it will help in reality because there is an admission controller that injects a secret volume into every pod.

This predicate should be a no-op if the pod does not use any PVCs.  @cofyc can you help check if there's anything we can fix in `FindPodVolumes` to return early?

Reply:
`FindPodVolumes` will return early if pod has no PVCs, see https://github.com/kubernetes/kubernetes/blob/5323aed97b5d7779930a71967ecd510e4b37cbb5/pkg/controller/volume/persistentvolume/scheduler_binder.go#L154-L178. 

How about skip `VolumeSchedulingStageLatency` logic and `klog.V` in `FindPodVolumes` too? This will make predicate a no-op if the pod does not use any PVCs.

Reply:
Thanks, @msau42 and @cofyc!
We definitely need to fix this issue. The current predicate is unacceptably slow for pods that do not request volumes.
`FindPodVolumes` gets pod name, reads time which often needs a system call, and increments Prometheus metrics even when a pod does not need any volume.
The predicate itself (which calls `FindPodVolumes`) also allocates an array for failure reasons and processes the output of FindPodVolumes for all pods whether they request a volume or not. The predicate itself must have a check to bypass all this and return immediately when a pod does not need a volume. I changed the PR to check PVCs and return immediately if there is no PVCs.
PTAL.